### ISSUE TITLE
.NET SDK projects seem to cause inefficiency in checking

### ISSUE THREAD

**Issue by dsyme** [Microsoft/visualfsharp#4608]
There is something fishy where .NET SDK project files (?) seem to cause some kind of inefficiency in VS project checking - perhaps repeated checking of files.

Compare these two similar scenarios:

1. open src\FSharp.Compiler.Service.fsproj (new-style projects), edit `src\fsharp\symbols\Symbols.fs`

against

1. open FSharp.sln (old-style projects), edit `src\fsharp\symbols\Symbols.fs`

The first takes 240seconds of CPU time to colorize.  The second takes about 40 seconds (as expected).

It *might* be because the project is checked twice - one for netstandard2.0 and one for net45. But I thought that wasn't implemented. That would explain only half the difference, and colorization shouldn't be delayed.

There is also a difference in treatment of signature files which I will also investigate.

**Comment by dsyme** [comment]
This is caused by the fact that `FSComp.fs` is not being reported in the list of valid files.  This causes a huge number of invalid-identifier errors to be reported, and a lot of time spent in preparing suggestions for names (cc @forki)

The list of files FCS is getting is here: https://gist.github.com/dsyme/db118033ddf720bd23c25c91489b3f81

**Comment by dsyme** [comment]
@cartermp I updated the description

**Comment by auduchinok** [comment]
> This is caused by the fact that FSComp.fs is not being reported in the list of valid files. This causes a huge number of invalid-identifier errors to be reported in the background, including a lot of time spent in the background preparing suggestions for names (cc @forki)

@dsyme I've checked what's coming from MSBuild invocation on the SDK-based project and files that are results of these targets do not appear there at all, there're only items that are passed to targets itself.

<img width="1046" alt="screen shot 2018-03-23 at 18 52 22" src="https://user-images.githubusercontent.com/3923587/37839547-c43d6410-2ecb-11e8-9c92-e7bc87b5cddc.png">

Checking older commit (before moving to SDK-based projects) shows these files evaluated as `CompileBefore` items.

<img width="1233" alt="screen shot 2018-03-23 at 18 26 46" src="https://user-images.githubusercontent.com/3923587/37839666-0a824d28-2ecc-11e8-872a-e82733e7fc26.png">

Checking current master if there're any `CompileBefore` items at all:
<img width="579" alt="screen shot 2018-03-23 at 18 48 43" src="https://user-images.githubusercontent.com/3923587/37839706-2afcbf5c-2ecc-11e8-99f1-c28d4a72860f.png">

**Comment by auduchinok** [comment]
> The corresponding generated file does get passed when using command line msbuild:

Probably additional targets should be triggered in IDEs, that aren't triggered now but triggered by MSBuild (e.g. if the build depends on it). It wasn't, however, needed on the old-style projects.
I'm looking what triggers them in console build.

**Comment by auduchinok** [comment]
OK, adding `GenerateCode` to the list of targets called on project load made `FSComp.fs` appear in Rider.

**Comment by cartermp** [comment]
Is this actually a performance problem? I'm not convinced it's leading to any slowdowns in the IDE. FSComp.fs is generated by design-time builds in VS and is in the `obj/` directory and gets recognized just fine. It's being multi-TFM that leads to any editor slowdowns.

**Comment by abelbraaksma** [comment]
What I did here https://github.com/dotnet/fsharp/pull/9918 maybe related. Also, the `#nowarn` in the generated files is ignored by the tooling (not by the compiler), leading to many redundant errors being generated, possibly adding to the slowdown. There's another open pr by me on the warnings where I intend to fix that (make sure that tooling recognizes the pragmas).

Though that won't explain all the perf issues, it may be causing some of it.

**Comment by cartermp** [comment]
Yeah i don't think the supposed perf issues are at all related to this issue
